prov/efa/src/rxr/rxr_pkt_type_req.c

```diff
@@ -751,6 +736,19 @@ ssize_t rxr_pkt_proc_matched_rtm(struct rxr_ep *ep,
 
 	assert(rx_entry->state == RXR_RX_MATCHED);
 
+	/* Adjust rx_entry->cq_entry.len as needed.
+	 * Initialy rx_entry->cq_entry.len is total recv buffer size.
+	 * rx_entry->total_len is from REQ packet and is total send buffer size.
+	 * if send buffer size < recv buffer size, we adjust value of rx_entry->cq_entry.len
+	 * also truncate rx_entry->iov to say memory registration.
+	 * if send buffer size > recv buffer size, we have a truncated message and will
+	 * write error CQ entry.
+	 */
+	if (rx_entry->cq_entry.len > rx_entry->total_len) {
+		rx_entry->cq_entry.len = rx_entry->total_len;
+		ofi_truncate_iov(rx_entry->iov, &rx_entry->iov_count, rx_entry->total_len);
```

Comment:
Does this not affect multi-recv buffers where we want to keep the full IOV in the posted rx_entry? It is also not clear to me how truncating the rx_entry IOV is reducing the amount of registered pages in this path.

Reply:
This happens after a rx-entry got matched, for multi-recv, this rx_entry is extracted from the master rx_entry, so I do not think it will affect multi-recv. The RX side memory registration happens only with the read message protocol, in which case, we have to register receiving buffer (local buffer), and when it is too large, we ran out of registered pages.

Reply:
Got it, thanks for the explanation. Might want to factor this into your comment then, that this is for the read path.

Reply:
Thank you! I think move the truncate to read path would be more efficient